app/react/Templates/components/MetadataTemplate.tsx

```diff
@@ -56,14 +65,29 @@ export class MetadataTemplate extends Component<MetadataTemplateProps> {
     this.onSubmitFailed = this.onSubmitFailed.bind(this);
   }
 
-  onSubmit = (_template: TemplateSchema) => {
+  onSubmit = async (_template: TemplateSchema) => {
     const template = { ..._template };
     template.properties = template.properties?.map(_prop => {
       const prop = { ..._prop };
       prop.label = _prop.label.trim();
       return prop;
     });
 
+    const mappingValidation = await validateMapping(template);
+    if (!mappingValidation.valid) {
+      const fields = mappingValidation.errors.map((e: any) => e.name);
+      this.context.confirm({
+        accept: () => {
+          template.reindex = true;
+          return this.props.saveTemplate(template);
+        },
+        title: 'Template conflict',
+        message: `The field or fields [${fields}] has been previously used for a different type of data,
```

Comment:
When singular and plural are mixed, the plural form must be maintained afterwards: "The field or fields"..."have been"